storm/src/test/storm/mesos/MesosNimbusTest.java

```diff
@@ -17,304 +17,451 @@
-    Map conf = new HashMap<>();
-    conf.put(MesosCommon.WORKER_CPU_CONF, 1);
-    conf.put(MesosCommon.WORKER_MEM_CONF, 1024);
-    conf.put(MesosCommon.EXECUTOR_CPU_CONF, 1);
-    conf.put(MesosCommon.EXECUTOR_MEM_CONF, 1024);
-    conf.put(MesosNimbus.CONF_EXECUTOR_URI, "");
-    mesosNimbus._conf = conf;
-
-    topologyMap.put("t1", new TopologyDetails("t1", conf, new StormTopology(), 5));
-    HashMap<OfferID, List<LaunchTask>> launchList = new HashMap<>();
-    HashMap<OfferID, List<WorkerSlot>> slotList = new HashMap<>();
-    slotList.put(offerId, Arrays.asList(new WorkerSlot("", 1000)));
-    Topologies topologies = new Topologies(topologyMap);
-
-    mesosNimbus.computeResourcesForSlot(
-        offers,
-        topologies,
-        launchList,
-        "t1",
-        slotList,
-        OfferID.newBuilder().setValue("derp").build()
-    );
-    assertEquals(1, launchList.size());
-    assertEquals(1, launchList.get(offerId).size());
-
-    assertEquals(
-        TestUtils.buildScalarResource("cpus", 1.0),
-        launchList.get(offerId).get(0).getTask().getResources(0)
-    );
-
-    assertEquals(0, offers.get(offerId).getResourcesCount());
+    Map<String, OfferResources> offerResourcesPerNode = MesosCommon.getConsolidatedOfferResourcesPerNode(rotatingMap);
+    Map<String, Collection<WorkerSlot>> workerSlotsMap = new HashMap<>();
+
+    Map<String,List<Protos.TaskInfo>> tasksToLaunch = mesosNimbus.getTasksToLaunch(topologies, workerSlotsMap, offerResourcesPerNode);
+
+    assertTrue(tasksToLaunch.isEmpty());
+  }
+
+  @Test
+  public void testGetTasksToLaunchForOneTopologyWithOneOffer() {
+    TopologyDetails t1 = TestUtils.constructTopologyDetails("t1",
+                                                            MesosNimbusTest.DEFAULT_WORKER_COUNT,
+                                                            MesosCommon.DEFAULT_WORKER_CPU,
+                                                            MesosCommon.DEFAULT_WORKER_MEM_MB);
+    Map m = t1.getConf();
+    Map<String, TopologyDetails> topologyDetailsMap = new HashMap<>();
+    topologyDetailsMap.put("t1", t1);
+
+    Topologies topologies = new Topologies(topologyDetailsMap);
+
+    // One offer with sufficient resources
+    Offer offer = TestUtils.buildOfferWithPorts("O-1", "h1", 24, 40000, 3100, 3200);
+    rotatingMap.put(offer.getId(), offer);
+
+    Map<String, OfferResources> offerResourcesPerNode = MesosCommon.getConsolidatedOfferResourcesPerNode(rotatingMap);
+    Map<String, Collection<WorkerSlot>> workerSlotsMap = new HashMap<>();
+    Collection<WorkerSlot> workerSlots = new ArrayList<>();
+    workerSlots.add(new WorkerSlot("h1", 3100));
+    workerSlotsMap.put("t1", workerSlots);
+
+    Map<String,List<Protos.TaskInfo>> tasksToLaunch = mesosNimbus.getTasksToLaunch(topologies, workerSlotsMap, offerResourcesPerNode);
+
+    assertTrue((tasksToLaunch.size() == 1) && (tasksToLaunch.get("h1").size() == 1));
```

Comment:
Why not do this as two separate `assertEquals` statements so that if it fails, it's more obvious which piece was the culprit?